### PR TITLE
Redesigning `protostar.toml` 7 — `ConfigurationFileFactory`

### PR DESCRIPTION
**Pull request by kasperski95**
`ConfigurationFileFactory` builds the proper version of the `ConfigurationFile`.

This PR doesn't change the Protostar logic yet. I'll merge this PR without review on 27 September at 3 pm.

### CODE REVIEW

File: protostar/configuration_file/configuration_file_factory.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+        raise ProtostarException(
+            f"{protostar_toml_path} must specify `protostar-version` for example:\n"
+            """
+            [project]
+            protostar-version = ?.?.?
+            """,
+        )
+
+    def _create_configuration_toml_v2(
```
Comment (MaksymilianDemitraszek) [inline protostar/configuration_file/configuration_file_factory.py:44]: Shouldn't we have `AbstractConfigFactory`, `V1ConfigFactory` and `V2ConfigFactory`, imo factory shouldn't be an entity responsible for making decisions.

Comment (MaksymilianDemitraszek) [reply protostar/configuration_file/configuration_file_factory.py:44]: My point is I wouldn't name it factory if you do not have a interchangeable family of those

**Review by MaksymilianDemitraszek — APPROVED** [review]
LGTM

File: protostar/configuration_file/configuration_file_factory.py (hunk 1 above)
Comment (kasperski95) [reply protostar/configuration_file/configuration_file_factory.py:44]: https://www.oreilly.com/library/view/head-first-design/0596007124/ch04.html
Chapter "Building a simple pizza factory"

In that chapter, the factory decides what Pizza needs to be build based on the provided type. In this case, the factory decides which configuration file should be created.

> My point is I wouldn't name it factory if you do not have a interchangeable family of those

Factory and Abstract Factory are two different things. In the link above, they use the Factory suffix even if the Factory is not abstract.

**Pull request merged**